src/schema.js

```diff
@@ -171,6 +181,16 @@ const blocks = {
 		labels: [null, DEFAULT_LABEL],
 		description: "Bundles gateway services accessible under a port.",
 	},
+	server_certificate: {
+		parents: (context) => {
+			if (context.length >= 2 && context[0].name === "tls" && context[1].name === "server") {
+				return ['tls']
+			}
+			return `"server_certificate" only valid in a "tls" in a "server" block.`
+		},
```

Comment:
As this repeats almost identically over and over, it might be worth a named reusable function. :recycle:

Reply:
hi @afflerbach 

> But, something's odd:
> ...

I cannot reproduce that.

Reply:
Good to hear that it's working for you. 

Here, for `tls` and its children, there's actually no duplication, but for all (?) other items. Then again TLS stuff is marked as error as mentioned above:

![Screenshot_20240409_081441](https://github.com/coupergateway/couper-vscode/assets/1311513/bb0686e0-f81f-44ad-857a-6f8f89cc1d3f)
![Screenshot_20240409_081425](https://github.com/coupergateway/couper-vscode/assets/1311513/6dfd42e6-d818-4c34-bed1-89234822b9bc)
![Screenshot_20240409_081548](https://github.com/coupergateway/couper-vscode/assets/1311513/4132a16d-be91-4e08-81df-9782cdb9c622)

Reply:
@afflerbach Just to make sure, which couper extension are you checking with? Couper Configuration v1.8.1?

Reply:
I'm on branch `135-tls`, the Couper extension from the store is not installed. 

I start VSCode with 

    $ code --extensionDevelopmentPath=$(pwd)
    
to get the latest changes. My last screenshot accordingly shows that the new `tls` attributes are suggested. 

    $ code --version
    1.88.0
    5c3e652f63e798a5ac2f31ffd0d863669328dc4c
    x64
    
:shrug:

Reply:
@afflerbach One quite weird thing: You have one tooltip content with "Documentation" and another with "Reference". "Reference" was changed to "Documentation" on Nov 21, 2022 with PR https://github.com/coupergateway/couper-vscode/pull/116 in commit https://github.com/coupergateway/couper-vscode/commit/b0d07f05cbb42535124d69b439baeda18ae1e840

Reply:
> one tooltip content with "Documentation" and another with "Reference". 
Hey, good catch, didn't see this.

I assumed Couper was not installed, because in the extensions window the green button reads **Install**.

**But:**

```
$ code --list-extensions --show-versions | grep couper
avengagermanygmbh.couper@1.5.1

$ code --uninstall-extension avengagermanygmbh.couper
Uninstalling avengagermanygmbh.couper...
Extension 'avengagermanygmbh.couper' was successfully uninstalled!
```

After deinstalling the legacy extension everything looks good!

I guess, that issue I had, occures to anybody who has both the old `couper` extension by `AvengaGermanyGmbH` and the new `couperconf` installed.